acceptance/tests/pxp-module-puppet/rerun_transaction.rb

```diff
@@ -0,0 +1,60 @@
+require 'pxp-agent/test_helper.rb'
+
+
+STATUS_QUERY_MAX_RETRIES = 60
+STATUS_QUERY_INTERVAL_SECONDS = 1
+
+test_name 'two runs with same transaction_id' do
+
+  step 'Ensure each agent host has pxp-agent running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config_json_using_puppet_certs(master, agent).to_s)
+      on agent, puppet('resource service pxp-agent ensure=running')
+      show_pcp_logs_on_failure do
+        assert(is_associated?(master, "pcp://#{agent}/agent"),
+               "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+      end
+    end
+  end
+
+  target_identities = agents.map {|agent| "pcp://#{agent}/agent"}
+  transaction_ids = []
+  run_results = []
+
+  step "run puppet to generate a transaction_id" do
+    transaction_ids = start_puppet_non_blocking_request(master, target_identities)
+    target_identities.zip(transaction_ids).each do |identity, transaction_id|
+      run_results << check_puppet_non_blocking_response(identity, transaction_id, STATUS_QUERY_MAX_RETRIES, STATUS_QUERY_INTERVAL_SECONDS, "unchanged")
+    end
+  end
+
+  step "restart pxp" do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config_json_using_puppet_certs(master, agent).to_s)
+      on agent, puppet('resource service pxp-agent ensure=running')
+      show_pcp_logs_on_failure do
+        assert(is_associated?(master, "pcp://#{agent}/agent"),
+               "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+      end
+    end
+  end
+
+  step "rerun with same transaction_id, expect error and status queryable" do
+    target_identities.zip(transaction_ids, run_results).each do |identity, transaction_id, first_run_result|
+      # This should fail!
+      response = rpc_request(master, [identity],
+                             'pxp-module-puppet', 'run',
+                             {:flags => ['--onetime',
+                                         '--no-daemonize',
+                                         '--environment', 'production']},
+                                         false,
+                                         transaction_id)[identity]
+
+      assert(response[:envelope][:message_type] == 'http://puppetlabs.com/rpc_provisional_response', 'Did not receive rpc provisional response')
```

Comment:
Can assert_equal be used here? That would have MiniTest tell us explicitly what the expected and actual message types were. Alternatively update the message to state the actual :message_type